javascript/react/basics-i/rendering-multiple-components.md

```diff
@@ -110,4 +111,7 @@ What is the name of the **identity attribute** that must be included when creati
 * `identity`
 * `ref`
 
-
+---
+## Footnotes
+[1:keys]
+To get a better understanding of this, let's take a look at how React decides when it should render or re-render a component. Knowing that in its rawest form, a React component is just an object with a set of properties, each time a change occurs in these properties React will compare the new and old versions and, if a difference is detected, it will redraw the component. If the defined `key` is not unique, the order of the components might get mixed which will lead to React thinking that all components have changed (instead of just one). This, in turn, will lead to React re-rendering all the components (instead of just the one that was changed) which might lead to performance issues. Take a look at [this article](https://dev.to/jtonzing/the-significance-of-react-keys---a-visual-explanation--56l7) if you want to learn more about this.
```

Comment:
We might wanna rephrase this one a bit more. We want the reader to know that the main reason a `key` is needed is for React to know the order of elements, and know when that order changes so that it can re-render their positions accordingly.

The reason we need a unique identifier for this is because nothing about a React element is immutable. We can change a `button` to an `input`, change the `id` of a `form`, whatever. This means that, between re-renders, React can't always know which element it is dealing with, so we add a unique key to help out React.

Reply:
Changed footnote to:

[1:keys]
To get a better understanding of this, let's take a look at how React decides when it should render or re-render a component. In its rawest form, a React component is just an object with a set of properties. Each time a change occurs in these properties, React will compare the new and old versions and, if a difference is detected, it will redraw the component. A unique `key` is needed so that React knows the order of elements. If the order changes, React will then know to to re-render the elements at their new positions.

If a unique `key` is not defined, between re-renders React won't know which element it is dealing with. This might lead to undesired re-renders or performance drops. To learn more about this check out this great article on [the significance of React keys](https://dev.to/jtonzing/the-significance-of-react-keys---a-visual-explanation--56l7).